### ISSUE TITLE
Support Project Loom in Jetty 10 & JDK 17+

### ISSUE THREAD

**Issue by ghost** [dropwizard/dropwizard#7134]
As of version 10, Jetty supports Project Loom and virtual threads

https://github.com/eclipse/jetty.project/issues/8007

We want to start porting our code to use Loom as soon as it goes GA.

Please add out of the box support for configuring Jetty with virtual threads in the standard HTTP Server config

**Comment by zUniQueX** [comment]
Before we can integrate this into the framework, we'd have to modify the `InstrumentedQueuedThreadPool` of `dropwizard-metrics` to support setting a thread factory.

**Comment by Akshay-Thakare** [comment]
+1

**Comment by ghost** [comment]
hi, we already have plans to start looking into Loom as soon as JDK 21 is released next month.
Any news on this? It would be a major issue if we were blocked from using Loom solely because of Dropwizard not supporting it :-/

Thank you, much appreciated

**Comment by zUniQueX** [comment]
Hi @jfurmankiewiczpros. I've just opened #7457 for basic loom support.

**Comment by ghost** [comment]
thank you, greatly appreciated

**Comment by zUniQueX** [comment]
Resolved by #7457 
Resolved by #7746

**Comment by dylanbaroody** [comment]
When is the target for 3.0.3 release?

**Comment by grnadav** [comment]
Was this also merged into 4.0.x ?
Any plans to upgrade Jetty to 12.x - where Virtual Threads are [enabled by default](https://webtide.com/jetty-12-virtual-threads-support/) ?

**Comment by jdvr** [comment]
>Was this also merged into 4.0.x ?

`enableVirtualThreads` option is available since 4.0.3

**Comment by zUniQueX** [comment]
@grnadav The upgrade to Jetty 12 won't be done for Dropwizard 3.x and 4.x but will be introduced in Dropwizard 5.

**Comment by grnadav** [comment]
@jdvr Thanks.
Saw the auto PR to upgrade to jetty v12 on v4 branch had issues (https://github.com/dropwizard/dropwizard/pull/7673)
The errors (https://github.com/dropwizard/dropwizard/actions/runs/6505578556/job/17669475383) seem to be caused by the servlet jar missing (not in the new bom?)

**Comment by grnadav** [comment]
@zUniQueX gotcha.
Anywhere I can track the progress towards 5.x ?

**Comment by zUniQueX** [comment]
@grnadav Jetty 12 extracted the servlet support from the server core and moved the old artifact to separte modules for different EE versions. That causes many breaking changes for the request handling. So we can't upgrade without breaking changes for our users and therefore defer the upgrade to our next major version.

We don't have a timeline for the release of Dropwizard 5 yet. Currently, the Jakarta EE 10 upgrade is blocked by the next release for the Jersey 3.1 branch. After this release is available, we can upgrade to EE 10 and move to a Java 17 baseline which will be the main change for Dropwizard 5.

**Comment by grnadav** [comment]
@zUniQueX dang thats a lot of deps lineup to track. good job with that!
Why is DW5 aiming for Java 17 and not 21? its LTS as well. More deps. issues?

**Comment by zUniQueX** [comment]
@grnadav We try to support the widest range of Java versions we can to make it easier for users to upgrade their applications in the same system environment. The previous Java baseline changes in Dropwizard were introduced because some libraries dropped support for older Java versions.

With Jetty 12 there is a minimum Java version of 17 required to run Jetty. Therefore we also use Java 17 as our baseline. But newer Java versions are supported as well and we'll run all pipelines against the newest LTS versions once they are released (as we did with Java 21).
